contracts/contracts/strategies/AerodromeEthStrategy.sol

```diff
@@ -0,0 +1,592 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title Aerodrome Automated Market Maker (AMO) Strategy
+ * @notice AMO strategy for the Aero OETH/ETH pool
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/Math.sol";
+
+import { IRouter } from "./../interfaces/aerodrome/IRouter.sol";
+import { IGauge } from "./../interfaces/aerodrome/IGauge.sol";
+import { IPool } from "./../interfaces/aerodrome/IPool.sol";
+import { IERC20, InitializableAbstractStrategy } from "../utils/InitializableAbstractStrategy.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { IWETH9 } from "../interfaces/IWETH9.sol";
+import { FixedPointMathLib } from "solady/src/utils/FixedPointMathLib.sol";
+
+contract AerodromeEthStrategy is InitializableAbstractStrategy {
+    using StableMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 public constant MAX_SLIPPAGE = 1e16; // 1%
+    address public constant ETH_ADDRESS =
+        0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    IGauge public immutable aeroGaugeAddress;
+    IRouter public immutable aeroRouterAddress;
+    address public immutable aeroFactoryAddress;
+
+    IPool public immutable lpTokenAddress;
+    IERC20 public immutable oeth;
+    IWETH9 public immutable weth;
+
+    // Ordered list of pool assets
+    uint128 public immutable wethCoinIndex;
+    uint128 public immutable oethCoinIndex;
+
+    // Maximum allowed imbalance percentage.
+    // Initially set as 5%, meaning the allowed range is 0.475 < ratio < 0.525
+    uint256 public poolImbalanceThreshold = 5_000; // 5%
+
+    // Used to circumvent the stack too deep issue
+    struct AerodromeEthConfig {
+        address aeroRouterAddress; //Address of the Aerodrome router
+        address aeroGaugeAddress; //Address of the Aerodrome gauge
+        address aeroFactoryAddress; // Address of the Aerodrome PoolFactory
+        address lpTokenAddress; //Address of the OETH/ETH Aero LPToken
+        address oethAddress; //Address of OETH token
+        address wethAddress; //Address of WETH
+    }
+
+    /**
+     * @dev Verifies that the caller is the Strategist.
+     */
+    modifier onlyStrategist() {
+        require(
+            msg.sender == IVault(vaultAddress).strategistAddr(),
+            "Caller is not the Strategist"
+        );
+        _;
+    }
+
+    /**
+     * @dev Checks the Aero sAMM pool's balances have improved and the balances
+     * have not gone beyond the threshold.
+     * The standard deposit function adds to both sides of the pool in a way that
+     * the pool's balance is not worsened.
+     * Withdrawals are proportional so doesn't change the pools asset balance.
+     */
+    modifier improvePoolBalance() {
```

Comment:
This method should check the pool's balances before the action that the modifier wraps, then compare them again after. 

This way we check that things got better during the action.

If I'm reading this correctly, this only compares the balances/ratios after the action has taken place, which means it will not be able to compare the before and after to see if the difference is better?

What is this modifier securing as written?

Reply:
> What is this modifier currently trying to do?
This modifier is used to make sure after rebalancing, the pool ratio doesn't skew more than the allowed threshold. As long as the ratio remains within threshold, we should be good. It helps avoid errors from the strategist while rebalancing the pool. This also gives the strategist the flexibility to rebalance the pool as per the market conditions. 

> This way we check that things got better during the action.
To validate that we should define what we mean by "better" for the pool. Does it mean the pool should be closer to 50:50 ratio or should it be unbalanced till the safest point.

Reply:
"Better" is towards 50/50.

Going towards 50/50 is safer because the strategist can't trade away from 50/50 which is an action that guaranteed loses money.

Going towards 50/50 is also more flexible, in that the strategist may want to move only a little towards repegging things. This can keep the price still off peg, but be profitable for the strategy, and slow down redeems.

Reply:
Fixed here: [7933c0e](https://github.com/OriginProtocol/origin-dollar/pull/2019/commits/7933c0e137c7efc9b826ae1bf7074337aa84163b)